src/main.ts

```diff
@@ -121,87 +143,110 @@ export class Matchy extends HTMLElement {
-                },
-                ["btn", "btn-primary"],
-                { click: () => this.uploadFile() }
-            )
-        );
         form.appendChild(
             createButton(
                 "Pre Submit",
                 {
                     id: ids.preSubmitFile,
                     type: button.button,
                 },
-                ["btn", "btn-primary"],
+                ["btn", "btn-primary", "mt-2", "m-2"],
                 { click:() => this.preSubmitFile() }
             )
         );
         form.appendChild(
             createButton(
                 "Submit",
-                { 
+                {
                     id: ids.submitFile,
                     type: button.button,
                 },
-                ["btn", "btn-success"],
+                ["btn", "btn-success", "mt-2", "m-2"],
                 { click: () => this.submitFile() }
             )
         );
         form.appendChild(
             createButton(
                 "Reset",
-                { 
+                {
                     id: ids.resetFile,
                     type: button.button,
                 },
-                ["btn", "btn-success"],
-                { 
+                ["btn", "btn-success", "mt-2", "m-2"],
+                {
                     click: () => {
                         this.uploadFile();
                         this.hideElementById(ids.resetFile);
-                        this.hideElementById(ids.submitFile); 
+                        this.hideElementById(ids.submitFile);
                     }
                 }
             )
         );
         container.appendChild(form);
         this.shadow.appendChild(container);
-        const table = createElement(tag.table, { id: ids.csvContent }, [bootstrap.table, "table-hover", "table-striped"]);
+        const table = createElement(tag.table, { id: ids.fileContent }, [bootstrap["table"], "table-hover", "table-striped"]);
         this.shadow.appendChild(table);
         this.hideElementById(ids.preSubmitFile);
         this.hideElementById(ids.submitFile);
         this.hideElementById(ids.resetFile);
+        this.initEventListeners();
+    }
+
+    initEventListeners() {
+        this.addEventListener('keydown', (event)=> {
+            // Check if the pressed key is Tab
+            if (event.key === 'Tab') {
+                if (this.currentSelectedcell  !== null) {
+                    this.selectNextCell();
+                    event.preventDefault();
+                }
+            }
+        });
+    }
+
+    selectNextCell() {
+        if (this.currentSelectedcell == null) {
+            return;
+        }
+        const currentCell = this.currentSelectedcell.cell;
+        let nextRowIndex = this.currentSelectedcell.rowIndex;
+        let nextColIndex = this.currentSelectedcell.colIndex;
+        if (nextColIndex < this.fileHeader.length - 1) {
+            nextColIndex++;
+        } else {
+            nextRowIndex++;
```

Comment:
if one or many lines were deleted, we will have gaps in the indexs so index++ may point to a deleted (no more existing line)